`11:33:33 AssertionError: False is not True : expected typename: '(a.MyStruct<a.Outer.CodingKeys>)' - actual typename: 'τ_0_0'`

@DougGregor Should we expect this to work after https://github.com/apple/swift/pull/22136 is merged?

--- a/packages/Python/lldbsuite/test/lang/swift/private_generic_self/TestSwiftPrivateGenericSelf.py
+++ b/packages/Python/lldbsuite/test/lang/swift/private_generic_self/TestSwiftPrivateGenericSelf.py
@@ -12,5 +12,4 @@
 import lldbsuite.test.lldbinline as lldbinline
 from lldbsuite.test.decorators import *
 
-lldbinline.MakeInlineTest(__file__, globals(), decorators=[swiftTest,
-    expectedFailureAll(bugnumber="rdar://problem/38231646")])
+lldbinline.MakeInlineTest(__file__, globals(), decorators=[swiftTest])

--- a/packages/Python/lldbsuite/test/lang/swift/private_generic_self/main.swift
+++ b/packages/Python/lldbsuite/test/lang/swift/private_generic_self/main.swift
@@ -18,7 +18,7 @@ struct MyStruct<S : MyKey> : MyProtocol {
 }
 extension MyProtocol {
   func decode() {
-        return //% lldbutil.check_variable(self, self.frame().FindVariable("self"),use_dynamic=True,use_synthetic=True,typename='(a.MyStruct<a.Outer.CodingKeys>)')
+        return //% lldbutil.check_variable(self, self.frame().FindVariable("self"),use_dynamic=True,use_synthetic=True,typename='a.MyStruct<a.Outer.CodingKeys>')
     }
 }
 

--- a/packages/Python/lldbsuite/test/make/Makefile.rules
+++ b/packages/Python/lldbsuite/test/make/Makefile.rules
@@ -528,7 +528,7 @@ endif
 # Set up variables to run the Swift frontend directly.
 #----------------------------------------------------------------------
 SWIFT=$(shell dirname $(SWIFTC))/swift
-SWIFT_FEFLAGS = $(shell echo $(patsubst -Xfrontend,,$(SWIFTFLAGS)) | sed -e 's/-Xlinker [^ ]*//g')
+SWIFT_FEFLAGS = $(shell echo $(patsubst -Xfrontend,,$(SWIFTFLAGS)) | sed -e 's/-Xlinker [^ ]*//g') -enable-anonymous-context-mangled-names
 SWIFT_FE=$(SWIFT) -frontend
 
 #----------------------------------------------------------------------

